### PR TITLE
Add back support for `ip` range aggregations.

### LINKED ISSUE #17700: IP range aggregation will not work anymore once points are integrated
The ip range aggregation uses the numeric representation of ip addresses to compute the ranges. However once points are integrated, ip addresses will be stored in doc values using a SORTED_SET (binary) representation. So we have two options:
 - remove the ip range aggregation and tell users to use filters aggregations instead
 - add support for range aggregations on SORTED_SET doc values

### PR DESCRIPTION
**Pull request by jpountz**
This commit adds support for range aggregations on `ip` fields. However it will
only work on 5.x indices.

Closes #17700

### CODE REVIEW

**Comment by jpountz** [comment]
> it will only work on 5.x indices.

In a follow-up PR, we could make it work on legacy indices too if we think it's worth doing. But I think this PR is a good start.

**Comment by clintongormley** [comment]
> In a follow-up PR, we could make it work on legacy indices too if we think it's worth doing. But I think this PR is a good start.

I think it would be worth doing.

File: core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/ip/IpRangeAggregatorBuilder.java [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+            }
+            this.key = key;
+            this.from = from;
+            this.to = to;
+        }
+
+        Range(String key, String mask) {
+            String[] splits = mask.split("/");
+            if (splits.length != 2) {
+                throw new IllegalArgumentException("Expected [ip/prefix_length] but got [" + mask + "], which contains more than one [/]");
```
Comment (colings86) [inline core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/ip/IpRangeAggregatorBuilder.java]: Maybe this message could be improved as it looks this exception would be thrown if the mask contains no `/` characters as well?

File: core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/ip/IpRangeAggregatorBuilder.java [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+            }
+            InetAddress value = InetAddresses.forString(splits[0]);
+            int prefixLength = Integer.parseInt(splits[1]);
+            // copied from InetAddressPoint.newPrefixQuery
+            if (prefixLength < 0 || prefixLength > 8 * value.getAddress().length) {
+                throw new IllegalArgumentException("illegal prefixLength [" + prefixLength
+                        + "]. Must be 0-32 for IPv4 ranges, 0-128 for IPv6 ranges");
```
Comment (colings86) [inline core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/ip/IpRangeAggregatorBuilder.java]: Is it worth printing the mask in this message too so the user can find which range is causing the error?

File: core/src/test/java/org/elasticsearch/search/aggregations/bucket/IpRangeIT.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+import java.util.Arrays;
+
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.search.aggregations.AggregationBuilders;
+import org.elasticsearch.search.aggregations.bucket.range.Range;
+import org.elasticsearch.test.ESIntegTestCase;
+
+@ESIntegTestCase.SuiteScopeTestCase
+public class IpRangeIT extends ESIntegTestCase {
```
Comment (colings86) [inline core/src/test/java/org/elasticsearch/search/aggregations/bucket/IpRangeIT.java]: Should we test script usage here, or does the IP agg not support scripts? (I didn't see anything to validate that scripts are not being used in the code)

File: docs/reference/aggregations/bucket/iprange-aggregation.asciidoc [hunk 4]
```diff
@@ -1,7 +1,7 @@
 [[search-aggregations-bucket-iprange-aggregation]]
-=== IPv4 Range Aggregation
+=== IP Range Aggregation
```
Comment (colings86) [inline docs/reference/aggregations/bucket/iprange-aggregation.asciidoc]: Do the Java API docs need updating for this change?

File: core/src/test/java/org/elasticsearch/search/aggregations/bucket/IpRangeIT.java (hunk 3 above)
Comment (jpountz) [inline core/src/test/java/org/elasticsearch/search/aggregations/bucket/IpRangeIT.java]: the values source parser is configured as not scriptable but I can add a test that we fail if a script is provided

**Comment by colings86** [comment]
I left a few minor comments but otherwise LGTM

**Pull request merged**